inc/issue.class.php

```diff
@@ -1503,4 +1504,162 @@ public static function getIssuesSummary(array $params = []) {
          'icon'  => $params['icon'],
       ];
    }
+
+   /**
+    * Get default join for legacy search engine (up to GLPI 10.0)
+    *
+    * @param string $itemtype
+    * @param string $ref_table
+    * @param array $already_link_tables
+    * @return string
+    */
+   public static function getDefaultJoinLegacy($itemtype, $ref_table, &$already_link_tables): string {
+      // Get default joins for tickets
+      $join = Search::addDefaultJoin(Ticket::getType(), Ticket::getTable(), $already_link_tables);
+      // but we want to join in issues
+      $join = str_replace('`glpi_tickets`.`id`', '`glpi_plugin_formcreator_issues`.`itemtype` = "Ticket" AND `glpi_plugin_formcreator_issues`.`items_id`', $join);
+      $join = str_replace('`glpi_tickets`', '`glpi_plugin_formcreator_issues`', $join);
+      $join = str_replace('`users_id_recipient`', '`requester_id`', $join);
+      if (Plugin::isPluginActive(PLUGIN_FORMCREATOR_ADVANCED_VALIDATION)) {
+         $join .= PluginAdvformCommon::addDefaultJoin($itemtype, $ref_table, $already_link_tables);
+      } else {
+         $issueSo = Search::getOptions($itemtype);
+         $join .= Search::addLeftJoin(
+            $itemtype,
+            $ref_table,
+            $already_link_tables,
+            $issueSo[9]['table'],
+            'users_id_validator',
+            0,
+            0,
+            $issueSo[9]['joinparams']
+         );
+         $join .= Search::addLeftJoin(
+            $itemtype,
+            $ref_table,
+            $already_link_tables,
+            $issueSo[11]['table'],
+            'users_id_validate',
+            0,
+            0,
+            $issueSo[11]['joinparams']
+         );
+         $join .= Search::addLeftJoin(
+            $itemtype,
+            $ref_table,
+            $already_link_tables,
+            $issueSo[16]['table'],
+            'groups_id_validator',
+            0,
+            0,
+            $issueSo[16]['joinparams']
+         );
+      }
+      return $join;
+   }
+
+   /**
+    * Get default join for search engine
+    *
+    * @param string $itemtype
+    * @param string $ref_table
+    * @param array $already_link_tables
+    * @return array
+    */
+   public static function getDefaultJoin($itemtype, $ref_table, &$already_link_tables): array {
+      $join = [];
+      // Get default joins for tickets
+      $ticket_joins = SQLProvider::getDefaultJoinCriteria(Ticket::getType(), Ticket::getTable(), $already_link_tables);
+      // but we want to join in issues
+      foreach ($ticket_joins['LEFT JOIN'] as $new_table => $ticket_join) {
```

Comment:
The ticket joins may be an empty array if the user can view all tickets. There should be a check that the 'LEFT JOIN' key exists.

Reply:
Yes, you're right